src/ui/public/doc_viewer/viewers/json.js

```diff
@@ -0,0 +1,27 @@
+import _ from 'lodash';
+import angular from 'angular';
+import 'ace';
+import docTableDetailViewsRegistry from 'ui/registry/doc_table_detail_views';
+
+import jsonHtml from './json.html';
+
+docTableDetailViewsRegistry.register(function () {
+  return {
+    title: 'JSON',
+    order: 20,
+    template: jsonHtml,
+    shouldShow: () => true,
```

Comment:
I have a few doc views that I only want to show for certain hits.  It looks at fields in the hit to determine if the doc view is useful or not (will it show something or will it just be a blank view).  That's why I added the shouldShow.